### PR TITLE
docs: create key concepts page for pull requests

### LINKED ISSUE #15797: Explain immortal PRs in new documentation page
### Describe the proposed change(s).

We don't explain what's an "immortal PR" and why we even have immortal PRs in the first place. When you search for "immortal" you get no results in the docs. This is confusing for our users.

This issue resolves discussion #15628.

Todos:

- [ ] Create new page for the concept of immortal PRs
  - [ ] Use term `immortal` from PR description in docs, so it shows up in search
  - [ ] Use `@rarkins`' explanation as a rough draft
  - [ ] Explain what causes a PR to be immortal
  - [ ] Explain how to fix immortal PRs
- [ ] Check out `recreateClosed` docs special cases, maybe incorporate in new page
  - [ ] Add link to new page
- [ ] Update default PR body text for the 👻 **Immortal:** section, it should have a link to the new page

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Create a Pull Requests key concepts page that explains:
  - why we have immortal PRs in the first place
  - that we use branch name and PR title as cache keys
  - how to get rid of immortal PRs
  - how Renovate finds existing PRs 
  - how Renovate behaves in case of immortal or non-immortal PRs
  - how Renovate can ignore PRs

## Context

Closes #15797.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/key-concepts/immortal-pull-requests.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Then the update becomes "Update react (major)", which is not safely ignorable, instead of "Update react to v16".
+
+## Future plans for immortal PRs
+
+In the future we may embed metadata in each PR identifying the exact files and packages + versions that PR contains.
+Then we could allow such PRs to be closed/ignored but then as soon as there's any chance to files/packages/versions being updated then we'd be cached busted and create a new PR.
+
+## How to fix immortal PRs
+
+...TODO...
```
Comment (HonkingGoose) [inline docs/usage/key-concepts/immortal-pull-requests.md:47]: @rarkins can you explain how to fix immortal PRs?

I guess the real fix is to get the unique version into the branch name and PR title, so the "cache key" is unique?

**Comment by rarkins** [comment]
I think we could make this page more generic to talk about Pull Request closing, reuse and immortality

**Comment by HonkingGoose** [comment]
> I think we could make this page more generic to talk about Pull Request closing, reuse and immortality

Sounds good to me.

@rarkins Can you first push a explanation on how to fix 👻 immortal PRs into my branch? After that's done, I'll make the page more generic by covering:

- [ ] PR closing
- [ ] PR re-use
- [ ] PR immortality

**Comment by rarkins** [comment]
I think we should have this page a bit more generic and not specific to immortal. e.g. explaining the theory of how Renovate locates existing PRs and how it behaves in case of immortal or non-immortal, ignoring PRs, etc.

**Comment by HonkingGoose** [comment]
@rarkins I've renamed the file, and created some new sections for the concepts you want to cover.

The next step is that one of the Renovate maintainers fills out my placeholder sections with the correct content. I don't know enough about Renovate's PR handling to do this right.

I'll stop pushing things, so that you maintainers can write new content and push it into my branch. 😉

**Comment by fregante** [comment]
Will PRs mentioning "Immortal PR" link to this page? The main piece of information I want to know is "How I can disable immortal PRs" which would currently be followed by "You can't"

**Comment by HonkingGoose** [comment]
> Will PRs mentioning "Immortal PR" link to this page? The main piece of information I want to know is "How I can disable immortal PRs" which would currently be followed by "You can't"

I like this idea, see issue:

- #16611

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**